mods/linked/scripts.js

```diff
@@ -0,0 +1,508 @@
+			break;
+		case 'shift': {
+			if (!action.pokemon.isActive) return false;
+			if (action.pokemon.fainted) return false;
+			action.pokemon.activeTurns--;
+			this.swapPosition(action.pokemon, 1);
+			for (const foeActive of action.pokemon.side.foe.active) {
+				if (foeActive.isStale >= 2) {
+					action.pokemon.isStaleCon++;
+					action.pokemon.isStaleSource = 'switch';
+					break;
+				}
+			}
+			break;
+		}
+
+		case 'beforeTurn':
+			this.eachEvent('BeforeTurn');
+			break;
+		case 'residual':
+			this.add('');
+			this.clearActiveMove(true);
+			this.updateSpeed();
+			this.residualEvent('Residual');
+			this.add('upkeep');
+			break;
+		}
+
+		// phazing (Roar, etc)
+		for (const pokemon of this.p1.active) {
+			if (pokemon.forceSwitchFlag) {
+				if (pokemon.hp) this.dragIn(pokemon.side, pokemon.position);
+				pokemon.forceSwitchFlag = false;
+			}
+		}
+		for (const pokemon of this.p2.active) {
+			if (pokemon.forceSwitchFlag) {
+				if (pokemon.hp) this.dragIn(pokemon.side, pokemon.position);
+				pokemon.forceSwitchFlag = false;
+			}
+		}
+
+		this.clearActiveMove();
+
+		// fainting
+
+		this.faintMessages();
+		if (this.ended) return true;
+
+		// switching (fainted pokemon, U-turn, Baton Pass, etc)
+
+		if (!this.queue.length || (this.gen <= 3 && ['move', 'residual'].includes(this.queue[0].choice))) {
+			// in gen 3 or earlier, switching in fainted pokemon is done after
+			// every move, rather than only at the end of the turn.
+			this.checkFainted();
+		} else if (action.choice === 'megaEvo' && this.gen >= 7) {
+			this.eachEvent('Update');
+			// In Gen 7, the action order is recalculated for a Pokémon that mega evolves.
+			const moveIndex = this.queue.findIndex(queuedAction => queuedAction.pokemon === action.pokemon && queuedAction.choice === 'move');
+			if (moveIndex >= 0) {
+				const moveAction = /** @type {MoveAction} */ (this.queue.splice(moveIndex, 1)[0]);
+				moveAction.mega = 'done';
+				this.insertQueue(moveAction, true);
+			}
+			return false;
+		} else if (this.queue.length && this.queue[0].choice === 'instaswitch') {
+			return false;
+		}
+
+		let p1switch = this.p1.active.some(mon => mon && !!mon.switchFlag);
+		let p2switch = this.p2.active.some(mon => mon && !!mon.switchFlag);
+
+		if (p1switch && !this.canSwitch(this.p1)) {
+			for (const pokemon of this.p1.active) {
+				pokemon.switchFlag = false;
+			}
+			p1switch = false;
+		}
+		if (p2switch && !this.canSwitch(this.p2)) {
+			for (const pokemon of this.p2.active) {
+				pokemon.switchFlag = false;
+			}
+			p2switch = false;
+		}
+
+		if (p1switch || p2switch) {
+			if (this.gen >= 5) {
+				this.eachEvent('Update');
+			}
+			this.makeRequest('switch');
+			return true;
+		}
+
+		this.eachEvent('Update');
+
+		return false;
+	},
+
+	pokemon: {
+		moveUsed: function (move, targetLoc) {
+			let lastMove = this.moveThisTurn ? [this.moveThisTurn, move] : move;
```

Comment:
I don't think this actually affects gameplay, but ``this.moveThisTurn`` is a string, while ``move`` is a ``Move`` instance, so this is a weird tuple.

If you are interested, you might want to normalize the types.